File: src/irrgarten/GameState.java

package irrgarten;

/**
 *
 * @author juan
 */
public class GameState {
    private String labyrinth;
    private String players;
    private String monsters;
    private int currentPlayer;
    private boolean winner;
    private String log;
    
    public GameState(String lab, String pl, String mons, int cpl, boolean win, 
                     String lg){
        labyrinth=lab;
        players=pl;
        monsters=mons;
        currentPlayer=cpl;
        winner=win;
        log=lg;
        
    }
    public String getLabyrinth(){
        return labyrinth;
    }
    public String getPlayers(){
        return players;
    }
    public String getMonsters(){
        return monsters;
    }
    public int getCurrentPlayer(){
        return currentPlayer;
    }
    public boolean getWinner(){
        return winner;
    }
    public String getLog(){
        return log;
    }
}

File: GameState.rb
module Irrgarten
  
  class GameState

    def initialize(labyrinth,players,monster,current_player,winner,log)
      @labyrinth=labyrinth
      @players=players
      @monsters=monster
      @current_player=current_player
      @winner=winner
      @log=log
    end

    attr_reader :labyrinth
    attr_reader :players
    attr_reader :monsters
    attr_reader :current_player
    attr_reader :winner
    attr_reader :log


    def labyrinth
      return @labyrinth
    end

    def players
      return @players
    end

    def monsters
      return @monsters
    end

    def current_player
      return @currentPlayer
    end

    def winner
      return @winner
    end

    def log
      return @log
    end
  end
end
